File: src/client/java/net/legitimoose/bot/discord/command/staff/Rejoin.java
package net.legitimoose.bot.discord.command.staff;

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import net.legitimoose.bot.LegitimooseBotClient;

public class Rejoin extends ListenerAdapter {
    @Override
    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
        if (!event.getName().equals("rejoin")) return;
        event.deferReply(true).queue();
        event.getHook().sendMessage("Rejoining server...").queue();
        LegitimooseBotClient.attemptRejoin(true);
    }
}

File: src/client/kotlin/net/legitimoose/bot/discord/staff/Rejoin.kt
package net.legitimoose.bot.discord.staff

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.legitimoose.bot.LegitimooseBotClient
import net.legitimoose.bot.discord.Command

class Rejoin(override val event: SlashCommandInteractionEvent) : Command {
  override fun onCommandReceived() {
    event.deferReply(true).queue()
    event.hook.sendMessage("Rejoining server...").queue()
    LegitimooseBotClient.rejoin(force = true)
  }
}
